c7n/resources/rds.py

```diff
@@ -2238,24 +2238,31 @@ class PendingMaintenance(Filter):
             resource: aws.rds
             filters:
               - pending-maintenance
+              - type: value
+                key: '"c7n:PendingMaintenance"[].PendingMaintenanceActionDetails[].Action'
+                op: intersect
+                value:
+                  - system-update
     """
 
+    annotation_key = 'c7n:PendingMaintenance'
     schema = type_schema('pending-maintenance')
     permissions = ('rds:DescribePendingMaintenanceActions',)
 
     def process(self, resources, event=None):
         client = local_session(self.manager.session_factory).client('rds')
 
         results = []
-        pending_maintenance = set()
+        resource_maintenances = {}
         paginator = client.get_paginator('describe_pending_maintenance_actions')
         for page in paginator.paginate():
-            pending_maintenance.update(
-                {action['ResourceIdentifier'] for action in page['PendingMaintenanceActions']}
-            )
+            for action in page['PendingMaintenanceActions']:
+                resource_maintenances.setdefault(action['ResourceIdentifier'], []).append(action)
 
         for r in resources:
-            if r['DBInstanceArn'] in pending_maintenance:
+            pending_maintenances = resource_maintenances.get(r['DBInstanceArn'], [])
+            if len(pending_maintenances) > 0:
```

Comment:
nit: a truthiness check would be sufficient here and possibly clearer

```suggestion
            if pending_maintenances:
```